source/components/newevent.jsx

```diff
@@ -0,0 +1,151 @@
+import React, { Component } from 'react'
+import { Input, Button, Modal, Form, Select, Message, Alert } from 'semantic-ui-react'
+import Notifications, {notify} from 'react-notify-toast';
+
+// import styles from './newevent.scss'
+
+import axios from 'axios'
+
+class NewEvent extends Component {
+
+    constructor() {
+        super();
+        this.state = {
+            name: '',
+            category: '',
+            points: 0,
+            date: '',
+            startTime: '',
+            endTime: '',
+            password: '',
+            showMsg: false,
+            message: '',
+        }
+    }
+
+    handleChange = (event, data) => {
+        console.log(`${event.target.id}: ${data.value}`)
+        this.setState({
+            [event.target.id]: data.value
+        })
+    }
+
+    // fixes off by one date error
+    formatDate = date => {
+        const newDate = new Date(date)
+        newDate.setDate(newDate.getDate() + 1)
+        return newDate
+    }
+
+    createEvent = async () => {
+        const date = this.formatDate(this.state.date)
+        const event = {
+            name: this.state.name,
+            category: this.state.category,
+            points: this.state.points,
+            date,
+            startTime: this.state.startTime,
+            endTime: this.state.endTime,
+            password: this.state.password,
+        }
+        const res = await axios.post('http://localhost:3000/api/events', event)
+        console.log(res)
+        if (res.data.code === 500) {
+            this.setState({
+                message: 'All fields are required.',
+                showMsg: true,
+            })
+        } else if (res.data.code === 404) {
+            this.setState({
+                message: 'Invalid password.',
+                showMsg: true,
+            })
+        } else {
+            this.props.toggleModal()
+            window.location.reload()
+        }
+    }
+
+    render() {
+        const categories = [
+            { key: 'c', text: 'Corporate', value: 'corporate' },
+            { key: 's', text: 'Social', value: 'social' },
+            { key: 'o', text: 'Outreach', value: 'outreach' },
+            { key: 'm', text: 'Mentoring', value: 'mentoring' },
+            { key: 't', text: 'Tech Team', value: 'techTeam' },
+            { key: 'g', text: 'General Meeting', value: 'generalMeeting' },
+            { key: 'ot', text: 'Other', value: 'other' }
+        ]
+        return (
+            <Modal open={this.props.open} onClose={this.props.toggleModal} closeIcon>
+                <Modal.Content>
+                    <Form onSubmit={this.createEvent} error={this.state.showMsg}>
```

Comment:
can we have field verification? like ensuring fields aren't empty and stuff
and also making sure numbers are numbers

Reply:
imo all the fields should be required except for `endTime`

for `endTime`, let's default to +1hr from start unless they specify it
we can include that info in the placeholder

Reply:
oh yea i actually handled empty fields on the backend after the user submits the form, idk if that's the best way to do it lol
i didn't check that numbers were numbers tho, i'll add that on the frontend

Reply:
we should also include on frontend like
highlighting field in red and showing tooltip or something underneath the field

Reply:
i tried to set endTime default but it's kinda hard cuz it's actually a string since i'm using the 'time' input type, and idk how i would calculate & display it after the user fills out the startTime cuz i would have to calculate it in the startTime onChange function and it's constantly changing lol

Reply:
ok that's fine then we can leave it for later

Reply:
probably do it in `onClick` before api request is sent. check if user put in anything for `endTime` and if not, then +1hr

Reply:
i'll also do this in a separate pr lol i feel like it might get a lil messy